gprofiler/metadata/application_identifiers_java.py

```diff
@@ -11,16 +11,19 @@
 from gprofiler.exceptions import CalledProcessError
 from gprofiler.log import get_logger_adapter
 from gprofiler.metadata.base_application_identifier import _ApplicationIdentifier
-from gprofiler.profilers.java import jattach_path
-from gprofiler.utils import run_process
+from gprofiler.profilers.java import JattachJcmdRunner
 
 _logger = get_logger_adapter(__name__)
 
 
 class _JavaJarApplicationIdentifier(_ApplicationIdentifier):
+    def __init__(self, jattach_jcmd_runner: JattachJcmdRunner):
+        self.jattach_jcmd_runner = jattach_jcmd_runner
+
     def get_app_id(self, process: Process) -> Optional[str]:
+        assert self.jattach_jcmd_runner is not None
```

Comment:
Is this assert needed? Your constructor assigns a non-optional `JattachJcmdRunner` to this field.

Reply:
one can pass `None` somehow, I its OK to assert

Reply:
You could tell that on every parameter to every function...

We don't add asserts in this case - it clutters the code. You can add asserts on public-facing APIs, or where mypy lacks info so an assert helps, or when it helps for code readability. Here's it's none of the cases, please remove